common/lib/src/data/course.dart

```diff
@@ -1,18 +1,22 @@
 import 'package:flashcards_common/src/data/data.dart';
 
 class CourseData extends Data {
+  final String id;
   final String name;
   final double progress;
 
-  CourseData(this.name, this.progress);
+//  todo implement creating new course with id
```

Comment:
yea, i know, the workflow is create empty object (`CourseData()`) and set/update member variables like name or progress. It was in some example, but i can't find it right now.

Reply:
https://github.com/flutter/plugins/tree/master/packages/cloud_firestore#usage

Reply:
Well, you create the object without the id (that is stored as a name of this object anyway), "push" changes or new object to firestore and if you have some streabuilder using the stream, it will load the data with the object data you have set + with the id aka name of the object. :-)